Specification project management proposal (#2621)
open-telemetry/opentelemetry-specification: .github/ISSUE_TEMPLATE/project.md

---
name: Project Tracking Issue
about: Proposals for large-scale specification work which requires a group effort.
labels: spec:project-tracking
---

### Description

Describe the goals, objectives, and requirements for the project. This include the motivations for starting the project now, as opposed to later.

### Project Board

> Once approved, a project should be managed using a github project board. This project board should be pre-populated with issues that cover all known deliverables, organized by timeline milestones.

Reviewer: nit:

```suggestion
Once approved, a project should be managed using a GitHub project board. This project board should be pre-populated with issues that cover all known deliverables, organized by timeline milestones.
```

> Once approved, a project should be managed using a GitHub project board. This project board should be pre-populated with issues that cover all known deliverables, organized by timeline milestones.

Reviewer 2: Given that these are large projects I think the TC as a whole needs to approve. We expect that projects like this are started infrequently so it is not a huge burden for the entire TC to discuss each project and make a decision. Given how much work we expect the project workgroup to put in to make a proposal for starting a project I think it is only fair that the TC pays its full attention to review the proposal. This review will also be a good opportunity to identify the sponsoring TC member(s).

> Once approved, a project should be managed using a GitHub project board. This project board should be pre-populated with issues that cover all known deliverables, organized by timeline milestones.

Author: There is one board for tracking all projects, called the Project tracking board. The board contains one tracking issue per project. Maybe it needs a better name?

Per the project template,part of starting work on a project would be to create a board for that particular project, and populate it with an initial set of issues and milestones.

> A TC member associated with the project can create the board, along with a new github label to automatically associate issues and PRs with the project. The project lead and all other relevant project members should have edit access to the board.

Reviewer: ```suggestion
A TC member associated with the project can create the board, along with a new GitHub label to automatically associate issues and PRs with the project. The project lead and all other relevant project members should have edit access to the board.
```

> A [Technical Committee](https://github.com/open-telemetry/community/blob/main/community-members.md#technical-committee) (TC) member associated with the project can create the board, along with a new GitHub label to automatically associate issues and PRs with the project. The project lead and all other relevant project members should have edit access to the board.

Author: Yes, especially if that means we can automate adding issues and PRs to a project's board by using a label.

> A [Technical Committee](https://github.com/open-telemetry/community/blob/main/community-members.md#technical-committee) (TC) member associated with the project can create the board, along with a new GitHub label to automatically associate issues and PRs with the project. The project lead and all other relevant project members should have edit access to the board.

Reviewer 2: ```suggestion
A [Technical Committee](https://github.com/open-telemetry/community/blob/main/community-members.md#technical-committee) (TC) member associated with the project can create the board, along with a new project-specific GitHub label to automatically associate issues and PRs with the project. The project lead and all other relevant project members should have edit access to the board.
```

Once created, please link to the project board here.

### Deliverables

A description of what this project is planning to deliver, or is in the process of delivering. This includes all OTEPs and their associated prototypes.

In general, OTEPs are not accepted unless they come with working prototypes available to review in at least two languages. Please discuss these requirements with a TC member before submitting an OTEP.

### Staffing / Help Wanted

Who is currently planning to work on the project? If a project requires specialized domain expertise, please list it here. If a project is missing a critical mass of people in order to begin work, please clarify.

#### Required staffing

Projects cannot be started until they the following participants have been identified:
* Every project needs a project lead, who is willing to bottom line the project and address any issues which are not handled by other project members.

> * At least two sponsoring TC members. TC sponsors are dedicated to attending meetings, reviewing proposals, and in general being aware of the state of the project and it’s technical details. TC sponsors guide the project through the spec process, keep the tracking issue up to date, and help to ensure that relevant community members provide input at the appropriate times.

Reviewer: nit:

```suggestion
* At least two sponsoring TC members. TC sponsors are dedicated to attending meetings, reviewing proposals, and in general being aware of the state of the project and it's technical details. TC sponsors guide the project through the spec process, keep the tracking issue up to date, and help to ensure that relevant community members provide input at the appropriate times.
```

> * Engineers willing to write prototypes in at least two languages (if relevant to project). Languages should be fairly different from each other (for example. Java and Python).

Reviewer: ```suggestion
* Engineers willing to write prototypes in at least two languages (if relevant to project). Languages should be fairly different from each other (for example: Java and Python).
```
